src/merkle/merkle_tree.rs

```diff
@@ -114,6 +114,28 @@ impl MerkleTree {
         Ok(path.into())
     }
 
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an iterator over the leaves of this [MerkleTree].
+    pub fn leaves(&self) -> core::slice::Iter<Word> {
+        let leaves_start = self.nodes.len() / 2;
+        self.nodes[leaves_start..].iter()
```

Comment:
One alternative is to use skip, being just a matter of style without perf gain

```rust
self.nodes.iter().skip(self.nodes.len() / 2)
```

Reply:
I like this better. I've actually changes this to:
```Rust
let leaves_start = self.nodes.len() / 2;
self.nodes.iter().skip(leaves_start).enumerate().map(|(i, v)| (i as u64, v))
```
To keep the interface consistent with `SimpleSmt::leaves()`.